src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -7,13 +7,220 @@
+static ucs_status_t uct_rdamcm_cm_ep_server_init(uct_rdmacm_cm_ep_t *cep,
+                                                 const uct_ep_params_t *params)
 {
     return UCS_ERR_NOT_IMPLEMENTED;
 }
 
+ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
+{
+    uct_rdmacm_cm_ep_t *cep = ucs_derived_of(ep, uct_rdmacm_cm_ep_t);
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+
+    if (rdma_disconnect(cep->id)) {
+        ucs_error("rdmacm_cm ep %p (id=%p) failed to disconnect from peer %p",
+                  cep, cep->id,
+                  ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                                   UCS_SOCKADDR_STRING_LEN));
+        return UCS_ERR_IO_ERROR;
+    }
+
+    ucs_debug("rdmacm_cm ep %p (id=%p) disconnecting from peer :%s", cep, cep->id,
+              ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                               UCS_SOCKADDR_STRING_LEN));
+    return UCS_OK;
+}
+
+uct_base_iface_t dummy_iface = {
+    .super = {
+        .ops = {
+            .ep_pending_purge = (void *)ucs_empty_function_return_success,
+            .ep_disconnect    = uct_rdmacm_cm_ep_disconnect,
+            .ep_destroy       = UCS_CLASS_DELETE_FUNC_NAME(uct_rdmacm_cm_ep_t)
+        }
+    }
+};
+
+UCS_CLASS_INIT_FUNC(uct_rdmacm_cm_ep_t, const uct_ep_params_t *params)
+{
+    ucs_status_t status;
+
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_CM)) {
+        ucs_print("No UCT_EP_PARAM_FIELD_CM set: %zu", params->field_mask);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS) ||
+        !(params->sockaddr_cb_flags & UCT_CB_FLAG_ASYNC)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (!(params->field_mask & (UCT_EP_PARAM_FIELD_SOCKADDR |
+                                UCT_EP_PARAM_FIELD_CONN_REQUEST))) {
+        ucs_print("Neithter UCT_EP_PARAM_FIELD_SOCKADDR or UCT_EP_PARAM_FIELD_CONN_REQUEST set: %zu",
+                 params->field_mask);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &dummy_iface);
+
+    self->cm                  = ucs_derived_of(params->cm, uct_rdmacm_cm_t);
+    self->wireup.priv_pack_cb = (params->field_mask &
+                                 UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) ?
+                                params->sockaddr_pack_cb : NULL;
+    self->disconnect_cb       = (params->field_mask &
+                                 UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB) ?
+                                params->disconnect_cb : NULL;
+    self->user_data           = (params->field_mask &
+                                 UCT_EP_PARAM_FIELD_USER_DATA) ?
+                                params->user_data : NULL;
+    self->cq                  = NULL;
+    self->qp                  = NULL;
+
+    if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR) {
+        status = uct_rdamcm_cm_ep_client_init(self, params);
+    } else {
```

Comment:
else if ( UCT_EP_PARAM_FIELD_CONN_REQUEST) {
  ...
} else {
  ucs_error("either ... or ... have to be provided in ...");
}